backend/tournesol/views/contributor_recommendations.py

```diff
@@ -0,0 +1,111 @@
+"""
+Overrides the Polls API for recommendations specific to one user
+"""
+from django.db.models import Case, F, Prefetch, Sum, When
+from django.shortcuts import get_object_or_404
+from rest_framework import serializers
+from rest_framework.permissions import IsAuthenticated
+
+from core.models import User
+from tournesol.serializers.contributor_recommendations import ContributorRecommendationsSerializer
+
+from ..models import ContributorRating, Entity, Poll
+from ..views import PollRecommendationsBaseAPIView
+
+
+class ContributorRecommendationsBaseView(PollRecommendationsBaseAPIView):
+    """
+    A base class used to factorize behaviours common to all contributor
+    recommendations views.
+    """
+
+    queryset = Entity.objects.none()
+    serializer_class = ContributorRecommendationsSerializer
+
+    def annotate_with_total_score(self, queryset, request, poll: Poll, user):
+        criteria_cases = []
+        for crit in poll.criterias_list:
+            url_weight = request.query_params.get(f"weights[{crit}]")
+            if url_weight is not None:
+                try:
+                    used_weight = int(url_weight)
+                except ValueError as error_value:
+                    raise serializers.ValidationError(
+                        f"Invalid weight value for criteria '{crit}'"
+                    ) from error_value
+            else:
+                used_weight = 10
+            criteria_cases.append(
+                When(
+                    contributorvideoratings__criteria_scores__criteria=crit,
+                    then=used_weight,
+                )
+            )
+        criteria_weight = Case(*criteria_cases, default=0)
```

Comment:
This part is duplicated is both poll views and contributor recommendations views.

Could be factorized in a method.